packages/compiler/src/init/init.ts

```diff
@@ -365,12 +447,29 @@ export class InitTemplateError extends Error {
 }
 
 function validateTemplateDefinitions(
-  templates: unknown,
-  file: SourceFile
-): asserts templates is Record<string, InitTemplate> {
-  const validator = createJSONSchemaValidator(InitTemplateDefinitionsSchema);
-  const diagnostics = validator.validate(templates, file);
-  if (diagnostics.length > 0) {
-    throw new InitTemplateError(diagnostics);
+  template: unknown,
+  templateName: SourceFile,
+  strictValidation: boolean
+): ValidationResult {
+  const validator = createJSONSchemaValidator(InitTemplateSchema, {
+    strict: strictValidation,
+  });
+  const diagnostics = validator.validate(template, templateName);
+  return { valid: diagnostics.length === 0, diagnostics };
+}
+
+function isFileSkipGeneration(fileName: string, files: InitTemplateFile[]): boolean {
+  for (const file of files) {
+    if (file.path === fileName) {
+      return file.skipGeneration;
+    }
   }
+  return false;
+}
+
+function logDiagnostics(diagnostics: readonly Diagnostic[]): void {
+  diagnostics.forEach((diagnostic) => {
+    // eslint-disable-next-line no-console
+    console.log(diagnostic.message);
```

Comment:
Do we consider the rest of the diagnostic, code/severity not useful here?

Reply:
Good suggestion